pool/paymentmgr.go

```diff
@@ -877,42 +943,66 @@ func (pm *PaymentMgr) payDividends(ctx context.Context, height uint32, treasuryA
 
 	// Ensure the wallet is aware of all the coinbase outputs being
 	// spent by the payout transaction.
-	var maxSpendableHeight uint32
-	for _, pmtSet := range pmts {
-		spendableHeight := pmtSet[0].EstimatedMaturity + 1
-		if maxSpendableHeight < spendableHeight {
-			maxSpendableHeight = spendableHeight
-		}
-	}
-
-	var stopAfter uint32
-	switch {
-	case maxSpendableHeight > height:
-		stopAfter = maxSpendableHeight - height
-	default:
-		stopAfter = 1
-	}
-
-	err = pm.confirmCoinbases(pCtx, inputTxHashes, stopAfter)
+	err = pm.confirmCoinbasesV2(pCtx, txB, inputTxHashes)
 	if err != nil {
-		atomic.AddUint32(&pm.failedTxConfs, 1)
-
-		// Track the transactions with inaccurate confirmation data.
-		pm.mtx.Lock()
-		for k, v := range inputTxHashes {
-			pm.txConfHashes[k] = v
+		if len(inputTxHashes) != 0 {
+			// Increment the failed tx confirmation counter.
+			atomic.AddUint32(&pm.failedTxConfs, 1)
+
+			// Track the transactions with inaccurate confirmation data.
+			pm.mtx.Lock()
+			for k, v := range inputTxHashes {
+				pm.txConfHashes[k] = v
+			}
+			pm.mtx.Unlock()
 		}
-		pm.mtx.Unlock()
 
-		// Do not error if coinbase spendable confirmation requests are
-		// terminated by the context cancellation.
+		// Do not error if coinbase spendable confirmation requests
+		// are terminated by the context cancellation.
 		if !errors.Is(err, errs.ContextCancelled) {
 			return err
 		}
 
 		return nil
 	}
 
+	// var maxSpendableHeight uint32
```

Comment:
If this PR is merged, we obviously need to remove all this commented code first (and any other code which is no longer required).

Reply:
Also need to remove `v2` from the name `confirmCoinbasesV2`

Reply:
yeah this isn't expected to be merged any time soon, the plan is to test this extensively to confirm it solves the inaccurate confirmation issue at least after a rescan. With that done I can remove the functions no longer needed, update the tests and get it ready for merging.